### PR TITLE
Add permission callback

### PR DESCRIPTION
**Pull request by HardeepAsrani**
This adds a permission callback to Rest request. @contactashish13

### CODE REVIEW

**Comment by HardeepAsrani** [comment]
@contactashish13 Can you check?

File: classes/Visualizer/Gutenberg/Block.php [hunk 1]
```diff
@@ -249,15 +258,19 @@ public function update_chart_data( $data ) {
...
+			update_post_meta( $data['id'], Visualizer_Plugin::CF_CHART_TYPE, $chart_type );
+			update_post_meta( $data['id'], Visualizer_Plugin::CF_SOURCE, $source_type );
 			update_post_meta( $data['id'], Visualizer_Plugin::CF_DEFAULT_DATA, $data['visualizer-default-data'] );
 			update_post_meta( $data['id'], Visualizer_Plugin::CF_SERIES, $data['visualizer-series'] );
 			update_post_meta( $data['id'], Visualizer_Plugin::CF_SETTINGS, $data['visualizer-settings'] );
 
 			if ( $data['visualizer-chart-url'] && $data['visualizer-chart-schedule'] ) {
-				update_post_meta( $data['id'], Visualizer_Plugin::CF_CHART_URL, $data['visualizer-chart-url'] );
-				apply_filters( 'visualizer_pro_chart_schedule', $data['id'], $data['visualizer-chart-url'], $data['visualizer-chart-schedule'] );
+				$chart_url = esc_url( $data['visualizer-chart-url'] );
```
Comment (contactashish13) [inline classes/Visualizer/Gutenberg/Block.php]: @HardeepAsrani I think this should be esc_url_raw

File: classes/Visualizer/Gutenberg/Block.php [hunk 2]
```diff
@@ -268,7 +281,8 @@ public function update_chart_data( $data ) {
 			}
 
 			if ( $data['visualizer-chart-url'] ) {
-				$content['source'] = $data['visualizer-chart-url'];
+				$chart_url = esc_url( $data['visualizer-chart-url'] );
```
Comment (contactashish13) [inline classes/Visualizer/Gutenberg/Block.php]: esc_url_raw

File: classes/Visualizer/Gutenberg/Block.php [hunk 3]
```diff
@@ -249,15 +258,19 @@ public function update_chart_data( $data ) {
...
 			update_post_meta( $data['id'], Visualizer_Plugin::CF_DEFAULT_DATA, $data['visualizer-default-data'] );
 			update_post_meta( $data['id'], Visualizer_Plugin::CF_SERIES, $data['visualizer-series'] );
 			update_post_meta( $data['id'], Visualizer_Plugin::CF_SETTINGS, $data['visualizer-settings'] );
 
 			if ( $data['visualizer-chart-url'] && $data['visualizer-chart-schedule'] ) {
-				update_post_meta( $data['id'], Visualizer_Plugin::CF_CHART_URL, $data['visualizer-chart-url'] );
-				apply_filters( 'visualizer_pro_chart_schedule', $data['id'], $data['visualizer-chart-url'], $data['visualizer-chart-schedule'] );
+				$chart_url = esc_url( $data['visualizer-chart-url'] );
+				update_post_meta( $data['id'], Visualizer_Plugin::CF_CHART_URL, $chart_url );
+				apply_filters( 'visualizer_pro_chart_schedule', $data['id'], $chart_url, $data['visualizer-chart-schedule'] );
```
Comment (contactashish13) [inline classes/Visualizer/Gutenberg/Block.php]: schedule can be sanitized as an integer

**Comment by HardeepAsrani** [comment]
@contactashish13 I've resolved all the current issues. I tested the import with two URLs,

- Google Sheet's CSV share URL
- A URL pointing to a csv file, like a http://www.example.com/chart.csv 

Let me know if that's enough to reproduce or you need more info. :)

**Pull request merged**